libs/utils/energy.py

```diff
@@ -325,4 +329,189 @@ def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
 
         return (channels_nrg, nrg_file)
 
+class ACME(EnergyMeter):
+    """
+    BayLibre's ACME board based EnergyMeter
+    """
+
+    def __init__(self, target, conf, res_dir):
+        super(ACME, self).__init__(target)
+
+        # Assume iio-capture is available in PATH
+        self._iiocapturebin = conf.get('iio-capture', 'iio-capture')
+        self._hostname = conf.get('ip_address', 'baylibre-acme.local')
+        self._channels = conf.get('channels', {
+            'CH0': '0'
+        })
+        self._iio = [None] * len(self._channels)
+
+        logging.info('%14s - ACME configuration:', 'ACME')
+        logging.info('%14s -     binary: %s', 'ACME', self._iiocapturebin)
+        logging.info('%14s -     device: %s', 'ACME', self._hostname)
+        logging.info('%14s -   channels:', 'ACME')
+        for channel in self._channels:
+            logging.info('%14s -      %s', 'ACME', self._str(channel))
+
+        # Check if iio-capture binary is available
+        try:
+            p = Popen([self._iiocapturebin, '-h'], stdout=PIPE, stderr=STDOUT)
+        except:
+            logging.error('%14s - iio-capture binary [%s] not available',
+                          'ACME', self._iiocapturebin)
+            logging.warning('\n\n'\
+                '  If you need to measure energy using an ACME EnergyProbe,\n'\
+                '  please do follow installation instructions available here:\n'\
+                '    https://github.com/ARM-software/lisa/wiki/Energy-Meters-Requirements#iiocapture---baylibre-acme-cape\n'\
+                '  Othwerwise, please select a different energy meter in your\n'\
+                '  configuration file'\
+            )
+            raise RuntimeError('Missing iio-capture binary')
+
+    def sample(self):
+        raise NotImplementedError('Not available for ACME')
+
+    def _iio_device(self, channel):
+        return 'iio:device{}'.format(self._channels[channel])
+
+    def _str(self, channel):
+        return '{} ({})'.format(channel, self._iio_device(channel))
+
+    def reset(self):
+        """
+        Reset energy meter and start sampling from channels specified in the
+        target configuration.
+        """
+        # Terminate already running iio-capture instance (if any)
+        wait_for_termination = 0
+        for proc in psutil.process_iter():
+            if self._iiocapturebin not in proc.cmdline():
```

Comment:
I get an error here because `proc.cmdline` is a `list` and it's not callable... I had to remove the parenthesis to make it work